eis_qgis_plugin/processing/algorithms/filtering/lee_additive_multiplicative_noise_filter.py

```diff
@@ -0,0 +1,86 @@
+from qgis.core import (
+    QgsProcessingParameterNumber,
+    QgsProcessingParameterRasterDestination,
+    QgsProcessingParameterRasterLayer,
+)
+
+from eis_qgis_plugin.processing.eis_processing_algorithm import EISProcessingAlgorithm
+
+
+class EISLeeAdditiveMultiplicativeNoiseFilter(EISProcessingAlgorithm):
+    def __init__(self) -> None:
+        super().__init__()
+
+        self._name = "lee_additive_multiplicative_noise_filter"
+        self._display_name = "Lee additive multiplicative noise filter"
+        self._group = "Filtering"
+        self._group_id = "filtering"
+        self._short_help_string = '''
+            Apply a Lee filter considering additive 
+            and multiplicative noise components to the input raster
+            '''
+
+    def initAlgorithm(self, config=None):
+        self.alg_parameters = [
+            "input_raster",
+            "size",
+            "add_noise_var",
+            "add_noise_mean",
+            "multi_noise_mean",
+            "output_raster",
+        ]
+
+        input_raster_param = QgsProcessingParameterRasterLayer(
+            name=self.alg_parameters[0], description="Input raster"
+            )
+        input_raster_param.setHelp("The input raster data set.")
+        self.addParameter(input_raster_param)
+
+        window_size_param = QgsProcessingParameterNumber(
+            name=self.alg_parameters[1],
+            description="Additive and Multiplicative Noise Variation",
```

Comment:
I'm assuming this is supposed to be the size param?

Reply:
Yes, GitHub Copilot did its tricks on me. I need to be more precise with the prompts.